src/Microsoft.HttpRepl/Program.cs

```diff
@@ -120,5 +121,16 @@ private static void ComposeDependencies(IConsoleManager consoleManager, out Http
                 { WellKnownPreference.JsonColor, "Green" }
             };
         }
+
+        private static HttpClient GetHttpClientWithPreferences(IPreferences preferences)
+        {
+            if (preferences.TryGetValue(HttpClientPreferences.UseDefaultCredentials, out var value) &&
```

Comment:
I'm somewhat surprised that this is the first preference we have that is a boolean, but it seems that it is. We should probably go ahead and add a `bool GetBoolValue(string preference, bool defaultValue)` to `IPreferences` and its implementations and use that here.